src/components/DetailedCard.tsx

```diff
@@ -0,0 +1,39 @@
+import { useSearchParams } from 'react-router';
+import { PersonWithoutUrl } from '../api/users.api';
+import Spinner from './Spinner/Spinner';
+
+export default function DetailedCard(props: {
+  data: PersonWithoutUrl;
+  isFetchPersonByIdLoading: boolean;
+  setCurrentCard: (currentCard: null) => void;
+}) {
+  const [searchParams, setSearchParams] = useSearchParams();
+
+  const handleCloseButtonClick = () => {
+    searchParams.set('id', '');
+    setSearchParams(searchParams);
+    props.setCurrentCard(null);
+  };
+
+  if (!props.data && !props.isFetchPersonByIdLoading) {
+    return;
+  }
+
+  return !props.isFetchPersonByIdLoading ? (
+    <div>
+      <h2>Detailed Card</h2>
+      <ul style={{ listStyleType: 'none', textAlign: 'start' }}>
+        {Object.entries(props.data).map((property, index) => {
```

Comment:
- try to use meaningful variables - just `property` is quite common and can be used for almost anything. In this case, at least it's better to have `persons` instead of just `data`. 
- additionally, good place for destructuring 
```suggestion
        {Object.entries(props.persons).map(([key, value], index) => {
```